packages/schematics/src/add/index.ts

```diff
@@ -28,180 +30,138 @@ if (fs.existsSync(join(__dirname, '../../package.json'))) {
-    // /[PROJECT_SRC]/[MODULE].module.ts
-    return normalize(modulePath + '.module.ts');
-  } else if (host.exists(modulePath + '/app/' + moduleBaseName)) {
-    // /[PROJECT_SRC]/app/[MODULE]
-    return normalize(modulePath + '/app/' + moduleBaseName);
-  } else if (host.exists(modulePath + '/app/' + moduleBaseName + '.ts')) {
-    // /[PROJECT_SRC]/app/[MODULE].ts
-    return normalize(modulePath + '/app/' + moduleBaseName + '.ts');
-  } else if (host.exists(modulePath + '/app/' + moduleBaseName + '.module.ts')) {
-    // /[PROJECT_SRC]/app/[MODULE].module..ts
-    return normalize(modulePath + '/app/' + moduleBaseName + '.module.ts');
-  } else {
-    throw new Error('Could not find the module, please specify a path to the app module file with the --module flag');
-  }
+// Chain a series of tasks to setup Clarity
+// 1. Add Clarity dependencies
+// 2. Add styles and scripts assets to angular.json
+// 3. Add ClarityModule to NgModule
+// 4. Add BrowserAnimationsModule to NgModule
+// 5. Run npm install
+export default function (options: NgAddOptions): Rule {
+  return chain([
+    setProjectSettings(options),
+    addClarityDependencies,
+    addAssetsToConfigFile,
+    importClarityModule,
+    importBrowserAnimationsModule,
+    runNpmInstall,
+  ]);
+}
+
+function setProjectSettings(options: NgAddOptions) {
+  return (host: Tree) => {
+    if (!host.exists(CONFIG_FILE_NAME)) {
+      throw new SchematicsException('Could not install Clarity, requires Angular and Angular CLI version 6 or greater');
+    }
+
+    projectSettings = { ...options };
+    const config = getJsonFile(host, CONFIG_FILE_NAME) as any;
+
+    if (!options.project) {
+      if (!config.defaultProject) {
+        throw new SchematicsException('Could not find a default project, please specify --project PROJECT_NAME');
+      }
+      projectSettings.project = config.defaultProject;
+    }
+
+    projectSettings.module = findAppModule(host, config, projectSettings.project || '');
+  };
 }
 
-// Read a file
-function readJsonFile(path: string) {
-  return JSON.parse(fs.readFileSync(path, 'utf-8'));
+function importClarityModule() {
+  return (host: Tree) => addModuleImportToModule(host, 'ClarityModule', '@clr/angular', projectSettings.module);
 }
 
-// Writes changes to a JSON file
-function updateJsonFile(path: string, callback: (a: any) => any) {
-  const json = readJsonFile(path);
-  callback(json);
-  fs.writeFileSync(path, JSON.stringify(json, null, 2));
+function importBrowserAnimationsModule() {
+  return (host: Tree) =>
+    addModuleImportToModule(
+      host,
+      'BrowserAnimationsModule',
+      '@angular/platform-browser/animations',
+      projectSettings.module
+    );
+}
+
+function addClarityDependencies(host: Tree) {
+  updateJsonFile(host, 'package.json', json => {
+    const version = getVersion(json.dependencies['@angular/core'], corePackage.version);
+
+    json.dependencies['@clr/angular'] = `${version}`;
```

Comment:
Is there a reason to use string literals. From what I'm seeing `getVersion` return already a string?